File: spC++/src/Implement/Models/Paragraph.cpp
#include "Headers/Models/Paragraph.hpp"

Paragraph::Paragraph(std::string text) {
    this->text = text;
}

Paragraph::Paragraph(Paragraph* paragraph) {
    this->text = paragraph->text;
}

IElement *Paragraph::clone() {
    return new Paragraph(this);
}

void Paragraph::accept(IVisitor* v) {
    v->visitParagraph(this);
}

std::string Paragraph::getText() {
    return this->text;
}

File: Java-Labs/lab/src/main/java/sp/lab/Models/Paragraph.java
package sp.lab.Models;

public class Paragraph implements IElement{
    private String text;

    public Paragraph(String text) {
        this.text = text;
    }

    public void print() {
        System.out.println("Paragraph: " + this.text);
    }

    public int getIndex() {
        return 1;
    }
}
